classes/form/CustomerForm.php

```diff
@@ -275,4 +282,35 @@ private function validateByModules()
             }
         }
     }
+
+    /**
+     * Performs validation on field values.
+     * Adds error to the field object if value is not as expected.
+     */
+    private function validateFieldsValues(): void
+    {
+        $this->validateFieldIsCustomerName('firstname');
+        $this->validateFieldIsCustomerName('lastname');
+    }
+
+    /**
+     * Checks whether a field's value is a valid customer(person) name.
+     *
+     * @param string $fieldName
+     */
+    private function validateFieldIsCustomerName(string $fieldName): void
```

Comment:
It would be great to be able to validate the behavior using tests. But it seems CustomerForm class rely heavily on static calls and singleton instances, so isolating it might be a very hard task?